cmake/extensions.cmake

```diff
@@ -2261,3 +2263,368 @@ function(target_byproducts)
                      COMMENT "Logical command for additional byproducts on target: ${TB_TARGET}"
   )
 endfunction()
+
+########################################################
+# 4. Zephyr devicetree function
+########################################################
+# 4.1. dt_*
+#
+# The following methods are for retrieving devicetree information in CMake.
+#
+# Note: In CMake we refer to the nodes using the node's path, therefore there
+# is no dt_path(...) function for obtaining a node identifier.
+
+# Usage:
+#   dt_nodelabel(<var> LABEL <label>)
+#
+# function for retrieving the node path for the node having nodelabel
+# <label>.
+#
+# the node path will be returned in the <var> parameter.
+#
+# <var>          : Return variable where the node path will be stored
+# LABEL <label>  : Node label
+function(dt_nodelabel var)
+  set(req_single_args "LABEL")
+  cmake_parse_arguments(DT_LABEL "" "${req_single_args}" "" ${ARGN})
+
+  if(${ARGV0} IN_LIST req_single_args)
+    message(FATAL_ERROR "dt_nodelabel(${ARGV0} ...) missing return parameter.")
+  endif()
+
+  foreach(arg ${req_single_args})
+    if(NOT DEFINED DT_LABEL_${arg})
+      message(FATAL_ERROR "dt_nodelabel(${ARGV0} ...) "
+                          "missing required argument: ${arg}"
+      )
+    endif()
+  endforeach()
+
+  get_target_property(${var} devicetree_target "DT_LABEL|${DT_LABEL_LABEL}")
+  if(${${var}} STREQUAL ${var}-NOTFOUND)
+    set(${var})
+  endif()
+
+  set(${var} ${${var}} PARENT_SCOPE)
+endfunction()
+
+# Usage:
+#   dt_node_exists(<var> NODE <node>)
```

Comment:
should this be `dt_path_exists(<var> PATH <path>)`

Reply:
I think `dt_node_exists` is a good name for the function. It should take `PATH <path>` for now, but we can extend it later to also accept `ALIAS <alias>`, `NODELABEL <label>`, etc. without changing the name. I think that would be nicer than having separate `dt_alias_exists` ,`dt_nodelabel_exists`, etc.